### PR TITLE
Dynamic information passing for ArrayLists and Resource usage

### PR DESCRIPTION
**Pull request by Tsathogguaa**
Resolved the issue with dynamic passage of arrays through intents from previous week. Also found and implemented a roundabout way of using resources such as ids and strings in test cases.

### CODE REVIEW

File: app/src/main/java/ch/sdp/vibester/activity/GameEndingActivity.kt [hunk 1]
```diff
@@ -8,53 +8,62 @@ import androidx.appcompat.app.AppCompatActivity
...
 class GameEndingActivity : AppCompatActivity() {
-    /*
-    private var statNames: Array<String> = arrayOf("")
-    private var statValues: Array<String> = arrayOf("")
-    private var incorrectSongs: Array<String> = arrayOf("")
-    */
 
+    private var incorrectSongs: ArrayList<String>? = arrayListOf("Default song")
+    private var statNames: ArrayList<String>? = arrayListOf("Default name 1", "Default name 2", "Default name 3", "Default name 4", "Default name 5")
+    private var statValues: ArrayList<String>? = arrayListOf("Default value 1", "Default value 2", "Default value 3", "Default value 4", "Default value 5")
```
[the remark this answers is not in the document]
Comment (Tsathogguaa) [reply app/src/main/java/ch/sdp/vibester/activity/GameEndingActivity.kt:14]: The statNames variable will be removed once we decide on a static array of stats.

File: app/src/main/java/ch/sdp/vibester/activity/GamescreenActivity.kt [hunk 2]
```diff
@@ -113,34 +113,30 @@ class GamescreenActivity: AppCompatActivity() {
...
         val intent = Intent(this, GameEndingActivity::class.java)
         //MOCK VALUES FOR INCORRECT SONGS, ADAPT FROM GAME DATA IN THE FUTURE
-        /*
-        val incArray: Array<String> = arrayOf("One", "Two", "Three")
-        val statNames: Array<String> = arrayOf("Hello there",
+        val incArray: ArrayList<String> = arrayListOf()
+        incArray.addAll(arrayOf("One", "Two", "Three"))
+
+        val statNames: ArrayList<String> = arrayListOf()
+        statNames.addAll(arrayOf("Hello there",
```
[the remark this answers is not in the document]
Comment (Tsathogguaa) [reply app/src/main/java/ch/sdp/vibester/activity/GamescreenActivity.kt:120]: The statNames variable will be removed once we decide on a static array of stats.

File: app/src/main/java/ch/sdp/vibester/activity/GamescreenActivity.kt [hunk 3]
```diff
@@ -113,34 +113,30 @@ class GamescreenActivity: AppCompatActivity() {
...
             "Second Stat",
             "Third Stat",
             "Fourth Stat",
-            "Fifth Stat")
-        val statRes: Array<String> = arrayOf("General Kenobi",
+            "Fifth Stat"))
+
+        val statVal: ArrayList<String> = arrayListOf()
+        statVal.addAll(arrayOf("General Kenobi",
```
[the remark this answers is not in the document]
Comment (Tsathogguaa) [reply app/src/main/java/ch/sdp/vibester/activity/GamescreenActivity.kt:127]: The statNames variable will be removed once we decide on a static array of stats.

File: app/src/main/java/ch/sdp/vibester/activity/GameEndingActivity.kt [hunk 4]
```diff
@@ -8,53 +8,62 @@ import androidx.appcompat.app.AppCompatActivity
...
 
 class GameEndingActivity : AppCompatActivity() {
-    /*
-    private var statNames: Array<String> = arrayOf("")
-    private var statValues: Array<String> = arrayOf("")
-    private var incorrectSongs: Array<String> = arrayOf("")
-    */
 
+    private var incorrectSongs: ArrayList<String>? = arrayListOf("Default song")
+    private var statNames: ArrayList<String>? = arrayListOf("Default name 1", "Default name 2", "Default name 3", "Default name 4", "Default name 5")
```
[the remark this answers is not in the document]
Comment (Tsathogguaa) [reply app/src/main/java/ch/sdp/vibester/activity/GameEndingActivity.kt:13]: The statNames variable will be removed once we decide on a static array of stats.

**Comment by Tsathogguaa** [comment]
As written in the codeclimate conversations, the duplicate code issue will be solved once we decide on a static array of stats, which will result in the removal of an ArrayList and thus, the duplicate code.

File: app/src/main/java/ch/sdp/vibester/activity/IncorrectSongsActivity.kt [hunk 5]
```diff
@@ -1,62 +1,63 @@
...
 import android.os.Bundle
 import android.view.Gravity
-import android.widget.Button
+import android.view.View
 import android.widget.LinearLayout
 import android.widget.LinearLayout.LayoutParams
 import android.widget.TextView
 import androidx.appcompat.app.AppCompatActivity
 import ch.sdp.vibester.R
-import ch.sdp.vibester.helper.IntentSwitcher
```
Comment (MaximeZmt) [inline app/src/main/java/ch/sdp/vibester/activity/IncorrectSongsActivity.kt:12]: maybe can add documentation here ? What does the class represent ? maybe method ?

File: app/src/main/java/ch/sdp/vibester/activity/GameEndingActivity.kt [hunk 6]
```diff
@@ -8,53 +8,62 @@ import androidx.appcompat.app.AppCompatActivity
 import ch.sdp.vibester.R
```
Comment (MaximeZmt) [inline app/src/main/java/ch/sdp/vibester/activity/GameEndingActivity.kt:9]: maybe can add documentation here ? What does the class represent ?

**Review by MaximeZmt — APPROVED** [review]
Good work, maybe consider adding some doc to class, otherwise LGTM ! :)

File: app/src/main/java/ch/sdp/vibester/activity/IncorrectSongsActivity.kt (hunk 5 above)
Comment (Tsathogguaa) [reply app/src/main/java/ch/sdp/vibester/activity/IncorrectSongsActivity.kt]: Done in the recent commit. Thanks for the feedback!

**Pull request merged**